### DISCUSSION TITLE
Migrate to latest available revision

### DISCUSSION THREAD

**Discussion by finswimmer** [sqlalchemy/alembic#949] [Usage Questions] [answered]
Hello everyone,

within my deployment pipeline, I want to make sure that the database is on the same revision as the latest available migration script.

If one is always go forward, this is quite easy by running `alembic upgrade head`. It's getting more complicated if I need to checkout an older version of my application, where I need to downgrade the database to the latest revision file in this older version.

The [cookbook](https://alembic.sqlalchemy.org/en/latest/cookbook.html#test-current-database-revision-is-at-head-s) and this [stackoverflow answer](https://stackoverflow.com/a/37093814/9750706) gives me an idea about how to get the latest revision number in the migration files and trigger update/downgrade.

As this seems to me like a quite common task, I wonder if there is already a complete solution out there (I couldn't find any until now)?

fin swimmer

**Comment by finswimmer** [comment]
Stupid me :man_facepalming: 

If the latest available migration script means a downgrade, there are of course no instructions about how to migrate. So rolling back needs some more advanced logic at all.

**Comment by zzzeek** [comment, marked as answer]
I think you need to downgrade the DB given the newer version of the code first, so the downgrade scripts are there, then check out the older version of the code.

in practice, for development databases people likely just rebuild dev databases from zero.  upgrades are much easier to deal with than downgrades.
